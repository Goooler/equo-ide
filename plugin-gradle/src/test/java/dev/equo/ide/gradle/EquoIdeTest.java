/*******************************************************************************
 * Copyright (c) 2022 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.ide.gradle;

import java.io.IOException;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

public class EquoIdeTest extends GradleHarness {
	@Test
	public void tasks() throws IOException {
		setFile("build.gradle").toLines("plugins { id 'dev.equo.ide' }", "equoIde {", "}");
		runAndAssert("tasks", "--stacktrace")
				.contains(
						"IDE tasks\n"
								+ "---------\n"
								+ "equoIde - Launches EquoIDE\n"
								+ "equoList - Lists the p2 dependencies of EquoIDE\n\n");
	}

	@Test
	public void p2repoArgCheck() throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://somerepo'",
						"  install 'org.eclipse.swt'",
						"}");
		runFailAndAssert("equoIde")
				.contains(
						"> Must end with /\n"
								+ "  p2repo(\"https://somerepo\")   <- WRONG\n"
								+ "  p2repo(\"https://somerepo/\")  <- CORRECT");
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://somerepo//'",
						"  install 'org.eclipse.swt'",
						"}");
		runFailAndAssert("equoIde")
				.contains(
						"> Must end with a single /\n"
								+ "  p2repo(\"https://somerepo//\")  <- WRONG\n"
								+ "  p2repo(\"https://somerepo/\")   <- CORRECT");
	}

	@Test
	public void equoIdeTestOnly() throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  p2repo 'https://download.eclipse.org/eclipse/updates/4.26/'",
						"  install 'org.eclipse.swt'",
						"}");
		runAndAssert("equoIde", "-PequoTestOnly=true", "--stacktrace")
				.contains("exit code: 0")
				.matches("(?s)(.*)stdout: Loaded (\\d+) bundles(.*)");
	}

	@Test
	@Disabled
	public void equoIde() throws IOException {
		setFile("build.gradle")
				.toLines(
						"plugins { id 'dev.equo.ide' }",
						"equoIde {",
						"  release '4.26'",
						"}",
						"// (placeholder so GPJ formats this nicely)");
		gradleRunner().withArguments("equoIde", "--stacktrace").forwardOutput().build();
	}
}
