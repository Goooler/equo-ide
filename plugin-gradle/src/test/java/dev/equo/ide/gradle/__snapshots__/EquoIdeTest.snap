dev.equo.ide.gradle.EquoIdeTest.help=[
Detailed task information for equoIde

Path
     :equoIde

Type
     EquoIdeTask (dev.equo.ide.gradle.EquoIdeTask)

Options
     --clean     Wipes all IDE settings and state before rebuilding and launching.

     --debug-classpath     Dumps the classpath (in order) without starting the application.
                           Available values are:
                                disabled
                                names
                                paths

     --debug-ide     The IDE will suspend until you attach a debugger.

     --init-only     Initializes the runtime to check for errors then exits.

     --show-console     Adds a visible console to the launched application.

     --use-atomos     Determines whether to use Atomos+Equinox or only Solstice's built-in OSGi shim

     --rerun     Causes the task to be re-run even if up-to-date.

Description
     Launches an Eclipse-based IDE for this project

Group
     IDE
]


dev.equo.ide.gradle.EquoIdeTest.p2repoArgCheck[double-slash]=[
.
> Must end with a single /
    https://somerepo//  <- WRONG
    https://somerepo/   <- CORRECT
]


dev.equo.ide.gradle.EquoIdeTest.p2repoArgCheck[no-slash]=[
.
> Must end with /
    https://somerepo   <- WRONG
    https://somerepo/  <- CORRECT
]


dev.equo.ide.gradle.EquoIdeTest.tasks=[
---------
equoIde - Launches an Eclipse-based IDE for this project
equoList - Lists the p2 dependencies of an Eclipse application
]